ambari-server/src/main/java/org/apache/ambari/server/controller/AddServiceRequest.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ambari.server.controller;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.Collections.emptySet;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.ambari.annotations.ApiIgnore;
+import org.apache.ambari.server.controller.internal.ProvisionAction;
+import org.apache.ambari.server.topology.ConfigRecommendationStrategy;
+import org.apache.ambari.server.topology.Configurable;
+import org.apache.ambari.server.topology.Configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+/**
+ * Data object representing an add service request.
+ */
+@ApiModel
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public final class AddServiceRequest {
+
+  static final String OPEATION_TYPE = "operation_type";
+  static final String CONFIG_RECOMMENDATION_STRATEGY = "config_recommendation_strategy";
+  static final String PROVISION_ACTION = "provision_action";
+  static final String STACK_NAME = "stack_name";
+  static final String STACK_VERSION = "stack_version";
+  static final String SERVICES = "services";
+  static final String COMPONENTS = "components";
+
+  private final OperationType operationType;
```

Comment:
Is this needed?  Will `operationType` in `AddServiceRequest` ever be something other than `ADD_SERVICE`, eg. `REMOVE_SERVICE`?  Wouldn't that request need a different domain object?

Reply:
I added it because it is in the specification. At least I need dummy getters and setters for this property. Will see in the future if MOVE_SERVICE and DELETE_SERVICE will have similar enough structure or need different domain objects.